solidity/random-beacon/test/RandomBeacon.Relay.test.ts

```diff
@@ -850,10 +872,569 @@ describe("RandomBeacon - Relay", () => {
+              it("should revert", async () => {
+                const { signatures, signingMembersIndices } =
+                  await signHeartbeatFailureClaim(
+                    members,
+                    0,
+                    group.groupPubKey,
+                    validFailedMembersIndices,
+                    groupThreshold
+                  )
+
+                await expect(
+                  randomBeacon.notifyFailedHeartbeat({
+                    groupId,
+                    failedMembersIndices: validFailedMembersIndices,
+                    // Remove the first signature to cause a mismatch with
+                    // the signing members count.
+                    signatures: `0x${signatures.slice(132)}`,
+                    signingMembersIndices,
+                  })
+                ).to.be.revertedWith("Unexpected signatures count")
+              })
+            }
+          )
+
+          context("when signatures count is less than group threshold", () => {
+            it("should revert", async () => {
+              const { signatures, signingMembersIndices } =
+                await signHeartbeatFailureClaim(
+                  members,
+                  0,
+                  group.groupPubKey,
+                  validFailedMembersIndices,
+                  // Provide one signature too few.
+                  groupThreshold - 1
+                )
+
+              await expect(
+                randomBeacon.notifyFailedHeartbeat({
+                  groupId,
+                  failedMembersIndices: validFailedMembersIndices,
+                  signatures,
+                  signingMembersIndices,
+                })
+              ).to.be.revertedWith("Too few signatures")
+            })
+          })
+
+          context("when signatures count is bigger than group size", () => {
+            it("should revert", async () => {
+              const { signatures, signingMembersIndices } =
+                await signHeartbeatFailureClaim(
+                  members,
+                  0,
+                  group.groupPubKey,
+                  validFailedMembersIndices,
+                  // All group signs.
+                  members.length
+                )
+
+              await expect(
+                randomBeacon.notifyFailedHeartbeat({
+                  groupId,
+                  failedMembersIndices: validFailedMembersIndices,
+                  // Provide one signature too much.
+                  signatures: signatures + signatures.slice(2, 132),
+                  signingMembersIndices: [
+                    ...signingMembersIndices,
+                    signingMembersIndices[0],
+                  ],
+                })
+              ).to.be.revertedWith("Too many signatures")
+            })
+          })
+        })
+      })
+
+      context("when failed members indices are incorrect", () => {
+        const assertFailedMembersIndicesCorrupted = async (
+          failedMembersIndices
```

Comment:
Let's declare the type for `failedMembersIndices`.
```suggestion
          failedMembersIndices: number[]
```

Reply:
See https://github.com/keep-network/keep-core/pull/2736/commits/7a6361d1e47a70083c05d702ad306f39d927f585